packages/core/src/components/tabs2/tabs2.tsx

```diff
@@ -150,9 +150,15 @@ export class Tabs2 extends AbstractComponent<ITabs2Props, ITabs2State> {
         }
     }
 
-    public componentDidUpdate(_prevProps: ITabs2Props, prevState: ITabs2State) {
+    public componentDidUpdate(prevProps: ITabs2Props, prevState: ITabs2State) {
         if (this.state.selectedTabId !== prevState.selectedTabId) {
             this.moveSelectionIndicator();
+        } else if (prevState.selectedTabId != null) {
+            const selectedTabChildren = this.getTabChildren();
+            const prevSelectedTabChildren = this.getTabChildren(prevProps.children);
+            if (selectedTabChildren !== prevSelectedTabChildren) {
```

Comment:
❌ this will _never_ be false because the two objects are always by definition different array instances.

better to check that the lengths are the same, but as you've pointed out there are still plenty of holes with that logic... not really sure the best approach here at the moment.